internal/cli/custom_domains.go

```diff
@@ -0,0 +1,319 @@
+				err := customDomainID.Pick(cmd, &inputs.ID, cli.customDomainsPickerOptions)
+				if err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			var customDomain *management.CustomDomain
+
+			if err := ansi.Waiting(func() error {
+				var err error
+				customDomain, err = cli.api.CustomDomain.Read(url.PathEscape(inputs.ID))
+				return err
+			}); err != nil {
+				return fmt.Errorf("Unable to get a custom domain with Id '%s': %w", inputs.ID, err)
+			}
+
+			cli.renderer.CustomDomainShow(customDomain)
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func createCustomDomainCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		Domain             string
+		Type               string
+		VerificationMethod string
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Args:  cobra.NoArgs,
+		Short: "Create a custom domain",
+		Long:  "Create a custom domain.",
+		Example: `auth0 branding domains create 
+auth0 branding domains create <id>`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := customDomainDomain.Ask(cmd, &inputs.Domain, nil); err != nil {
+				return err
+			}
+
+			customDomain := &management.CustomDomain{
+				Domain: &inputs.Domain,
+			}
+
+			if len(inputs.Type) > 0 {
+				customDomain.Type = apiProvisioningTypeFor(inputs.Type)
+			} else {
+				customDomain.Type = auth0.String(customDomainProvisioningTypeAuth0)
+			}
+
+			if len(inputs.VerificationMethod) > 0 {
+				customDomain.VerificationMethod = apiVerificationMethodFor(inputs.VerificationMethod)
+			}
+
+			if err := ansi.Waiting(func() error {
+				return cli.api.CustomDomain.Create(customDomain)
+			}); err != nil {
+				return fmt.Errorf("An unexpected error occurred while attempting to create the custom domain '%s': %w", inputs.Domain, err)
+			}
+
+			cli.renderer.CustomDomainCreate(customDomain)
+			return nil
+		},
+	}
+
+	customDomainDomain.RegisterStringU(cmd, &inputs.Domain, "")
+	customDomainType.RegisterStringU(cmd, &inputs.Type, "")
+	customDomainVerification.RegisterStringU(cmd, &inputs.VerificationMethod, "")
+
+	return cmd
+}
+
+func deleteCustomDomainCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		ID string
+	}
+
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Delete a custom domain",
+		Long:  "Delete a custom domain.",
+		Example: `auth0 branding domains delete 
+auth0 branding domains delete <id>`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				err := customDomainID.Pick(cmd, &inputs.ID, cli.customDomainsPickerOptions)
+				if err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			if !cli.force && canPrompt(cmd) {
+				if confirmed := prompt.Confirm("Are you sure you want to proceed?"); !confirmed {
+					return nil
+				}
+			}
+
+			return ansi.Spinner("Deleting custom domain", func() error {
+				_, err := cli.api.CustomDomain.Read(url.PathEscape(inputs.ID))
+
+				if err != nil {
```

Comment:
is "not found" the only source of errors there?

Reply:
We only do that GET to find out if the resource exists. Granted, it can fail because other reasons as well (e.g. network error), but unfortunately, we have no way to tell them apart. The delete call does not fail if the resource didn't exist, it ends successfully. That might be the worse outcome in this case.
What do you suggest?

Reply:
Maybe rework the error message to mention that there are other potential reasons?

Reply:
yes, maybe just remove the _The specified Id: %v doesn't exist_ and add the error msg?

Reply:
Done in https://github.com/auth0/auth0-cli/pull/311/commits/8f932f292c4a17940a5545e89d4ef6539b544212